src/parser/priest/shadow/modules/talents/ShadowCrash.tsx

```diff
@@ -2,39 +2,47 @@ import React from 'react';
 
 import SPELLS from 'common/SPELLS/index';
 import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
-import Events, { DamageEvent } from 'parser/core/Events';
+import Events, { DamageEvent, EnergizeEvent } from 'parser/core/Events';
 import Statistic from 'interface/statistics/Statistic';
 import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
 import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
 
 import ItemDamageDone from 'interface/ItemDamageDone';
 import { formatNumber } from 'common/format';
 import AbilityTracker from 'parser/priest/shadow/modules/core/AbilityTracker';
+import Insanity from 'interface/icons/Insanity'
 
 // Example Log: /report/zgBQ3kr6aAv19MXq/22-Normal+Zul+-+Kill+(2:26)/3-Selur
 class ShadowCrash extends Analyzer {
   static dependencies = {
     abilityTracker: AbilityTracker,
   };
+
   damage: number = 0;
+  insanityGained: number = 0;
   totalTargetsHit: number = 0;
   protected abilityTracker!: AbilityTracker;
 
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(SPELLS.SHADOW_CRASH_TALENT.id);
-    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.SHADOW_CRASH_TALENT_DAMAGE), this.onShadowCrashDamage);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.SHADOW_CRASH_TALENT_DAMAGE), this.onDamage);
+    this.addEventListener(Events.energize.by(SELECTED_PLAYER).spell(SPELLS.SHADOW_CRASH_TALENT), this.onEnergize);
   }
 
   get averageTargetsHit() {
     return this.totalTargetsHit / this.abilityTracker.getAbility(SPELLS.SHADOW_CRASH_TALENT.id).casts;
   }
 
-  onShadowCrashDamage(event: DamageEvent) {
+  onDamage(event: DamageEvent) {
     this.totalTargetsHit += 1;
     this.damage += event.amount;
```

Comment:
make sure you are including absorbed damage as well

`this.damage += event.amount + {event.absorb || 0);`